core/management/commands/migrate.py

```diff
@@ -0,0 +1,100 @@
+import json
+import os
+import sys
+from uuid import uuid4
+from functools import lru_cache
+
+from cachier import cachier
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.core.management.base import BaseCommand
+import requests
+
+from core.models import Approach, Submission, Task, Team
+from core.tasks import _field_file_to_local_path
+from django.contrib.auth.models import User
+from django.db.transaction import commit
+
+
+def use_first_nonself_arg(*args, **kwargs):
+    return args[0][1]
+
+
+class Command(BaseCommand):
+    help = 'Migrate 2016 data from Covalic'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--data', help='',
+        )
+
+    @cachier(hash_params=use_first_nonself_arg)
+    def girder_user_from_id(self, user_id):
+        r = requests.get(
+            f'https://challenge.kitware.com/api/v1/user/{user_id}',
+            headers={
+                'Girder-Token': os.getenv('GIRDER_TOKEN')
+            },
+        )
+
+        r.raise_for_status()
+        return r.json()
+
+    def _maybe_create_user(self, creator_id, creator_name):
+        girder_user = self.girder_user_from_id(creator_id)
+        user_obj = User.objects.get(email=girder_user['email'])
+
+        if not user_obj:
+            user_obj, _ = User.objects.create(
+                username=user['email'][:150],
+                password='dummy',
+                email=user['email'],
+                first_name=user['first_name'][:30],
+                last_name=user['last_name'][:150],
+                is_active=True,
+                date_joined=datetime.utcfromtimestamp(user['created']['$date'] / 1000),
+            )
+            user_obj.set_unusable_password()
+            user_obj.save()
+
+        return user_obj
+
+    def _create_team(self, user, uniqueid):
+        obj, _ = Team.objects.get_or_create(
+            challenge_id=34, creator=user, name=f'<autogenerated {uniqueid}>'
```

Comment:
For 2016 (i.e. when we have no other data fields available), I'd make the team `name` the submission title.